Assignment/HW5_SQL/HW5_Starter_Code/app/models.py

```diff
@@ -1,13 +1,111 @@
 import sqlite3 as sql
 
-def insert_data():
+# ========== UPDATE DATABASE FUNCTIONS ==========
+
+# insert a customer data record into the customers db,
+# returns the customer ID
+def insert_customer(fname, lname, company, email, phone):
+    # SQL statement to insert into database goes here
+    with sql.connect("app.db") as con:
+        cur = con.cursor()
+
+        cur.execute(
+        "INSERT INTO customers (first_name, last_name, company, email, phone) \
+        VALUES (?, ?, ?, ?, ?)", (fname, lname, company, email, phone))
+
+        customer_id = cur.lastrowid
+        con.commit()
+        return customer_id
+
+# insert an address to the address db
+# returns the address ID
+def insert_address(street, city, state, country, zip_code):
+    # SQL statement to insert into database goes here
+    with sql.connect("app.db") as con:
+        cur = con.cursor()
+
+        cur.execute(
+        "INSERT INTO addresses (street_address, city, state, country, zip_code)\
+        VALUES (?, ?, ?, ?, ?)", (street, city, state, country, zip_code))
+
+        address_id = cur.lastrowid
+        con.commit()
+        return address_id
+
+# insert a customer_address relationship to the customer_address table
+def insert_customer_address(customer_id, address_id):
+    # SQL statement to insert into database goes here
+    with sql.connect("app.db") as con:
+        cur = con.cursor()
+
+        cur.execute(
+        "INSERT INTO customer_address (customer_id, address_id) \
+        VALUES (?, ?)", (customer_id, address_id))
+

```

Comment:
Interesting use of an additional table to store the relationship between the address and the customer. Any particular rationale for this ?